v3/src/components/graph/adornments/lsrl/lsrl-adornment-component.tsx

```diff
@@ -327,21 +335,34 @@ export const LSRLAdornment = observer(function LSRLAdornment(props: IProps) {
       )
     }
 
-  }, [cellKey, classFromKey, confidenceBandPaths, containerId, dataConfig, equationContainerClass, fixEndPoints,
-      getLines, handleHighlightLineAndEquation, handleMoveEquation, instanceKey, model, plotHeight, plotWidth,
-      showConfidenceBands, xAxis, xCellCount, xScale, xSubAxesCount, yAxis, yCellCount, yScale, ySubAxesCount])
-
-  // Refresh values on axis changes
+  }, [cellKey, classFromKey, containerId, dataConfig, equationContainerClass, fixEndPoints, getLines,
+      handleHighlightLineAndEquation, handleMoveEquation, model, plotHeight, plotWidth, showConfidenceBands,
+      updateConfidenceBands, updateEquations, xAxis, yAxis])
+
+  // Refresh values on interceptLocked change
+  useEffect(function refreshInterceptLockChange() {
+    const updateCategoryOptions = graphModel.getUpdateCategoriesOptions()
+    return mstReaction(
+      () => {
+        model.updateCategories(updateCategoryOptions)
+      },
+      () => {
+        buildElements()
+      }, { name: "LSRLAdornmentComponent.refreshInterceptLockChange" }, model)
+  }, [buildElements, dataConfig, graphModel, interceptLocked, model, updateLSRL, xAxis, yAxis])
+
+  // Refresh values on changes to axes
   useEffect(function refreshAxisChange() {
-    return mstAutorun(() => {
-      // We observe changes to the axis domains within the autorun by extracting them from the axes below.
-      // We do this instead of including domains in the useEffect dependency array to prevent domain changes
-      // from triggering a reinstall of the autorun.
-      const { domain: xDomain } = xAxis // eslint-disable-line @typescript-eslint/no-unused-vars
-      const { domain: yDomain } = yAxis // eslint-disable-line @typescript-eslint/no-unused-vars
-      updateLSRL()
-    }, { name: "LSRLAdornmentComponent.refreshAxisChange" }, model)
-  }, [dataConfig, model, xAxis, yAxis, updateLSRL])
+    return mstAutorun(
+      () => {
+        // We observe changes to the axis domains within the autorun by extracting them from the axes below.
+        // We do this instead of including domains in the useEffect dependency array to prevent domain changes
+        // from triggering a reinstall of the autorun.
+        const { domain: xDomain } = xAxis // eslint-disable-line @typescript-eslint/no-unused-vars
+        const { domain: yDomain } = yAxis // eslint-disable-line @typescript-eslint/no-unused-vars
+        buildElements()
```

Comment:
Based on the lengthy discussion above:
```suggestion
        graphModel.getUpdateCategoriesOptions()
        buildElements()
```